### PR TITLE
Serilog & scripting updates

### PR DESCRIPTION
**Pull request by baughj**
## Description

* Remove log4net and replace with Serilog.

* Begin cleanup of logging. Logs now go to separate files in `./logs` in the data directory, and
  only general logs appear in the console. Over time more logging statements will be updated to
  use Serilog's structured property format. 

* Add globally accessible logger `GameLog` with shims for converting existing log4net log
  lines, and adding new log support for Serilog.

* Hybrasyl now shuts down correctly (and logs shutdown) when you close its console window,
  as opposed to just terminating instantly.

* Re-add missing assembly info.

* Fix some extremely subtle bugs related to Dialog / DialogSequences that surfaced from
  scripting (associate set incorrectly; jumpdialogs not always working)

* After we process a dialog option - stop further processing of the dialog packet.
  This fixes a number of strange bugs when navigating through dialogs.

## Related PRs
n/a

## Checklist
- [x] Tested and working locally
- [ ] Reviewed
- [ ] Tested and working on dev server
- [ ] Deployed to staging

## How to QA
Run Hybrasyl.
Note that temple attendants work as expected.

## Impacted Areas in Hybrasyl
Scripting dialogs
Logging

### CODE REVIEW

File: hybrasyl/Dialogs.cs [hunk 1]
```diff
@@ -21,22 +21,19 @@
  */
 
 using Hybrasyl.Objects;
-using log4net;
-using System;
```
Comment (norrismiv) [inline hybrasyl/Dialogs.cs]: pls readd `System`, unexpected results can occur when removing.

File: hybrasyl/Game.cs [hunk 2]
```diff
@@ -191,36 +217,47 @@ public static HybrasylConfig GatherConfig()
...
+
+            // Configure logging 
+
+            // Default is info
+            LevelSwitch = new LoggingLevelSwitch();
+            LevelSwitch.MinimumLevel = LogEventLevel.Information;
+
+            // We log every LogType defined in our enumeration to its own file. Only the "general" type is sent to the console.
+            var log = new LoggerConfiguration().MinimumLevel.ControlledBy(LevelSwitch).Enrich.WithThreadId().Enrich.WithExceptionData().
+                WriteTo.Map("LogType", "General", (name, wt) => wt.File($"{Path.Combine(Constants.DataDirectory, "logs")}/log-{name}.log", rollingInterval: RollingInterval.Day)).
```
Comment (norrismiv) [inline hybrasyl/Game.cs]: does rolling interval maintain previous logs with a datestamp attached? if no, that'd probably be beneficial.

Comment (norrismiv) [reply hybrasyl/Game.cs]: may also want to set a maximum logfile size/length.

File: hybrasyl/Dialogs.cs (hunk 1 above)
Comment (baughj) [reply hybrasyl/Dialogs.cs]: lol wut

File: hybrasyl/Game.cs (hunk 2 above)
Comment (baughj) [reply hybrasyl/Game.cs]: Yes, it maintains them, no removal is done.
The logs end up like: `log-General20190823.log` though now that I think about it they could just be called `General-20190823.log` so i'll update that

File: hybrasyl/Dialogs.cs (hunk 1 above)
Comment (norrismiv) [reply hybrasyl/Dialogs.cs]: in rare occasions, even where something isn't referenced directly from the base `System` class, unexpected results can occur due to the namespace not being included. it's a very old .net bug. not sure if it applies to core still

Comment (baughj) [reply hybrasyl/Dialogs.cs]: fixed

File: hybrasyl/Game.cs (hunk 2 above)
Comment (baughj) [reply hybrasyl/Game.cs]: Name updated, rotated automatically if size exceeds 1gb, 90 days of logs kept

**Comment by norrismiv** [comment]
👍

**Pull request merged**